relays/lib-substrate-relay/src/conversion_rate_update.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Tools for updating conversion rate that is stored in the runtime storage.
+
+use relay_utils::metrics::F64SharedRef;
```

Comment:
Conversion rate updater is an optional component - bridge may work without it (like Rialto<>Millau has worked before) or we may choose other option. Metrics values, on the other hand, are updated even if metrics endpoint is disabled (`--no-prometheus`), so even if we'll move to some other metrics kind or remove metrics at all, this value will still be updated internally.

We may split into three components - one who's preparing values, one that reads values and turns them into metrics and another last - conversion rate updater. But I have no strong belief that it is required. Please, log an issue if you do :)

Reply:
> Metrics values, on the other hand, are updated even if metrics endpoint is disabled 

Okay fair enough, I thought that it will simply stop working.

> We may split into three components - one who's preparing values, one that reads values and turns them into metrics and another last - conversion rate updater. 

Yeah, that sounds a bit complicated, but would probably be the cleanest. That said though, I wonder if having standalone metrics is really useful for anything (i.e without conversion updater, just metrics for the sake of metrics, like we have now). Anyways, it's not critical and I guess it doesn't even make sense to pollute the issue tracker.

Reply:
> Yeah, that sounds a bit complicated, but would probably be the cleanest. That said though, I wonder if having standalone metrics is really useful for anything (i.e without conversion updater, just metrics for the sake of metrics, like we have now). Anyways, it's not critical and I guess it doesn't even make sense to pollute the issue tracker.

Disabled metrics usefulness is questionable, ofc :) But they're printed in logs, so still may be useful to someone who's trying to see what happens. Initially they were collected on every relay loop iteration because calling `metrics.update()` is a bit easier than `if let Some(ref metrics) = metrics { metrics.update() }` and the overhead is minimal.